scripts/process-data/fitbitsleeplogs.R

```diff
@@ -214,23 +215,78 @@ sleeplogsdetails_df <-
   arrow::open_dataset(file.path(downloadLocation, "dataset_fitbitsleeplogs_sleeplogdetails")) %>% 
   select(all_of(sleeplogsdetails_vars)) %>% 
   collect() %>% 
-  left_join(y = (df %>% select(LogId, IsMainSleep)), by = "LogId") %>% 
-  group_by(LogId) %>% 
+  distinct() %>% 
+  left_join(y = (df %>% select(ParticipantIdentifier, LogId, IsMainSleep)), 
+            by = join_by("LogId", "ParticipantIdentifier")) %>% 
+  group_by(ParticipantIdentifier, LogId, id) %>% 
```

Comment:
what is the 'id' column here?

Reply:
I am using `id` in combination with `ParticipantIdentifier` and `LogId` as a redundancy for my grouping keys.

I believe `id` and `index` are added to the dataset by D&T's ETL pipeline- each unique combination of `ParticipantIdentifier` and `LogId` is assigned a group number (`id`) and each row in each `id`-numbered group is assigned a unique `index`.